rules/Naming/Rector/Class_/RenamePropertyToMatchTypeRector.php

```diff
@@ -158,12 +162,30 @@ private function renamePropertyPromotion(ClassLike $classLike): void
                 return;
             }
 
+            if ($this->isNodeOrType($param)) {
+                continue;
+            }
+
             $desiredPropertyNames[$key] = $desiredPropertyName;
         }
 
         $this->renameParamVarName($classLike, $constructClassMethod, $desiredPropertyNames);
     }
 
+    private function isNodeOrType(Param $param): bool
+    {
+        $type = $param->type;
+        if ($type instanceof NullableType) {
+            $type = $type->type;
+        }
+
+        return $type instanceof FullyQualified && $this->nodeTypeResolver->isObjectTypes($type, [
+            new ObjectType('PhpParser\Node'),
+            new ObjectType('PHPStan\Type\Type'),
+            new ObjectType('Ergebnis\Json\Printer\PrinterInterface'),
+        ]);
+    }
```

Comment:
These value should not be hardcoded. The solution must be generic, otherwise we'll pile up list of ignored clases here.
What is the goal here?

Reply:
That list of skipped change param types : `Node`, PHPStan Type, and `Ergebnis `'s `PrinterInterface`, do you have idea to make it generic?

Reply:
This should is already implemented in other type to name rules. There might be some inspiration.

In the end there should be 1 service that suggests a name based on type for all type to name rule.

Reply:
Do you mean `RenameParamToMatchTypeRector` ?

https://github.com/rectorphp/rector-src/blob/e50ad80290438afb4da04d92104b7d1d8ed9213e/rules/Naming/Rector/ClassMethod/RenameParamToMatchTypeRector.php#L115-L117

that seems not handle `Node`, or `PHPStan Type`, or `Ergebnis 's PrinterInterface`. There is seems a `Ramsey UUID` Interface check there:

https://github.com/rectorphp/rector-src/blob/e50ad80290438afb4da04d92104b7d1d8ed9213e/rules/Naming/Guard/BreakingVariableRenameGuard.php#L228-L231

should be added more there?

Reply:
I'm looking into this...